### PR TITLE
Add support for releasing package archives to GitHub

### LINKED ISSUE #3: Support deploying to GitHub
This should be able to create GitHub releases and upload package tarballs to those releases. It should also be able to auto-generate release notes from the existing changelog.

### PR DESCRIPTION
**Pull request by nex3**
Closes #3

### CODE REVIEW

File: lib/github.dart [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+    log("Released $pkgHumanName $pkgVersion to GitHub.");
+  }
+}
+
+/// A regular expression that matches a Markdown code block.
+final _codeBlock = RegExp(" *```");
+
+/// Returns the most recent section in the CHANGELOG, reformatted to remove line
+/// breaks that will show up on GitHub.
+String _lastChangelogSection() {
```
Comment (srawlins) [inline lib/github.dart]: A lot of this method seems fragile, but you don't want to write a full markdown parser :/

E.g. I can see some projects using h1 (`#`); there might be less than 1 space or more  than one space  after `##`. Probably not too important until you get some bugs filed 😄

File: lib/github.dart [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+/// Returns the GitHub repo name from the pubspec's `homepage` field.
+String _repoFromPubspec() =>
+    pubspec.homepage == null ? null : _parseHttp(pubspec.homepage);
+
+/// Parses a GitHub repo name from an SSH reference or a `git://` URL.
+///
+/// Returns `null` if it couldn't be parsed.
+String _parseGit(String url) {
+  var match =
+      RegExp(r"^(git@github\.com:|git://github\.com/)([^/]+)/([^/]+?)(\.git)?$")
```
[the remark this answers is not in the document]
Comment (nex3) [reply lib/github.dart]: Done. I also simplified the regexp a bit.

File: lib/github.dart [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+String _pkgGithubPassword;
+
+/// Returns the HTTP basic authentication Authorization header from the
+/// environment.
+String get _authorization =>
+    "Basic ${base64.encode(utf8.encode("$pkgGithubUser:$pkgGithubPassword"))}";
+
+/// The Markdown-formatted release notes to use for the GitHub release.
+///
+/// By default, this looks for a `CHANGELOG.md` file at the root of the
```
[the remark this answers is not in the document]
Comment (nex3) [reply lib/github.dart]: Yes, but the user can change the contents (derived by whatever logic they want) by setting this field.

File: lib/github.dart (hunk 1 above)
Comment (nex3) [reply lib/github.dart]: Yeah, given that users have the escape hatch of manually setting `pkgGithubReleaseNotes`, I think it's fine to wait on making this super customizable.

**Pull request merged**